Add April 2026 Flink community update blog post (#839)
apache/flink-web: docs/content/posts/2026-04-13-community-update.md

* **[FLIP-558](https://cwiki.apache.org/confluence/display/FLINK/FLIP-558%3A+Improvements+to+SinkUpsertMaterializer+and+changelog+disorder): Improvements to SinkUpsertMaterializer and Changelog Disorder.** - Addresses correctness issues in upsert materialisation under out-of-order changelog events.
* **[FLIP-557](https://cwiki.apache.org/confluence/display/FLINK/FLIP-557%3A+Granular+Control+over+Data+Reprocessing+in+Materialized+Table+Evolution): Granular Control over Data Reprocessing in Materialized Table Evolution.** - Gives users finer control over which partitions are reprocessed when a materialized table definition changes.
* **[FLIP-551](https://cwiki.apache.org/confluence/display/FLINK/FLIP-551%3A+Make+FRESHNESS+Optional+for+Materialized+Tables): Make FRESHNESS Optional for Materialized Tables.** (targeting Flink 2.2) - Reduces the boilerplate required when defining materialized tables where freshness guarantees are not needed.

### FLIPs (under discussion)

> * **[FLIP-202](https://cwiki.apache.org/confluence/display/FLINK/FLIP-202): [DRAFT] Introduce ClickHouse Connector.** - Community discussion ongoing for an officially supported ClickHouse sink connector.

Reviewer: these 3 flips do not seem to have had march activity

* **[FLIP-558](https://cwiki.apache.org/confluence/display/FLINK/FLIP-558%3A+Improvements+to+SinkUpsertMaterializer+and+changelog+disorder): Improvements to SinkUpsertMaterializer and Changelog Disorder.** - Addresses correctness issues in upsert materialisation under out-of-order changelog events.
* **[FLIP-557](https://cwiki.apache.org/confluence/display/FLINK/FLIP-557%3A+Granular+Control+over+Data+Reprocessing+in+Materialized+Table+Evolution): Granular Control over Data Reprocessing in Materialized Table Evolution.** - Gives users finer control over which partitions are reprocessed when a materialized table definition changes.
* **[FLIP-551](https://cwiki.apache.org/confluence/display/FLINK/FLIP-551%3A+Make+FRESHNESS+Optional+for+Materialized+Tables): Make FRESHNESS Optional for Materialized Tables.** (targeting Flink 2.2) - Reduces the boilerplate required when defining materialized tables where freshness guarantees are not needed.

### FLIPs (under discussion)

* **[FLIP-202](https://cwiki.apache.org/confluence/display/FLINK/FLIP-202): [DRAFT] Introduce ClickHouse Connector.** - Community discussion ongoing for an officially supported ClickHouse sink connector.

> * **[FLIP-332](https://cwiki.apache.org/confluence/display/FLINK/%5BWIP%5DFLIP-332%3A+Introduce+the+concept+of+state+self-sustained): [WIP] Introduce State Self-Sustained Concept.** - Explores state backends capable of managing their own lifecycle independently of checkpoints.

Reviewer 2: why do we have it at all?

> * **[FLIP-267](https://cwiki.apache.org/confluence/display/FLINK/FLIP+267%3A+Iceberg+Connector): Iceberg Connector.** - Continuing effort to formalise the Iceberg integration as a first-class Flink connector.

Reviewer 2: why do we have it at all here?
FLIP-267: last modified Oct 2022 https://cwiki.apache.org/confluence/display/FLINK/FLIP+267%3A+Iceberg+Connector
discussion ML last response October 17, 2022 8:21:17 PM GMT+2: https://lists.apache.org/thread/ycdll6wxj4npz5h7zvoh9khtfw8vy9nr
voting ML last response October 25, 2022 8:06:00 AM GMT+2 https://lists.apache.org/thread/0lchkmf1g5z6lqbz8hmr5685tt0nlh8v

---
authors:
- jlpayne72: null
  name: Julian Payne
date: "2026-04-13T08:00:00Z"
excerpt: Flink Community update for April 2026
title: Flink Community update for April 2026
aliases:
- /news/2026/04/13/community-update.html
---

This is the monthly Flink Community update for April 2026. We scour the latest updates from the Flink community so you don't have to! We'll dive into a few hot topics and FLIPs (Flink Improvement Proposals) before providing the usual almanac-style listing of technical updates, user-facing updates, and governance and community.
<!--more-->

<!-- TOC -->
  * [Hot Topics](#hot-topics)
    * [Apache Flink 2.3 Release Draws Nearer](#apache-flink-23-release-draws-nearer)
    * [Flink CDC 3.6.0 Release](#flink-cdc-360-release)

>     * [Flink Documentation Re-structure](#flink-documentation-re-structure)
>     * [Flink Native S3 FileSystem Connector](#flink-native-s3-filesystem-connector)
>   * [Developer/Technical Updates](#developertechnical-updates)
>     * [Table API & SQL](#table-api--sql)
>     * [Runtime & Execution](#runtime--execution)
>     * [Checkpointing & State](#checkpointing--state)
>     * [Metrics & Observability](#metrics--observability)
>     * [Python & Multi-language Support](#python--multi-language-support)
>   * [User-Facing Updates](#user-facing-updates)
>     * [Apache Flink CDC 3.6.0 Release](#apache-flink-cdc-360-release-1)
>     * [Apache Flink Agents 0.2.1 Release](#apache-flink-agents-021-release)
>     * [Documentation](#documentation)
>   * [Governance and Community](#governance-and-community)
>     * [FLIP Activity](#flip-activity)
>       * [Accepted - being implemented](#accepted---being-implemented)
>       * [Being discussed](#being-discussed)
>     * [Apache Flink Dev List activity](#apache-flink-dev-list-activity)
>   * [For more information](#for-more-information)
> <!-- TOC -->

Reviewer 3: ```suggestion
<!-- TOC -->
  * [Hot Topics](#hot-topics)
  * [Developer/Technical Updates](#developertechnical-updates)
  * [User-Facing Updates](#user-facing-updates)
  * [Governance and Community](#governance-and-community)
<!-- TOC -->
```

Personally, I'd trim this down to just one level

> Previous Blog: [Flink community update March 2026](https://flink.apache.org/2026/03/01/flink-community-update-for-march-2026/)

Reviewer 3: ```suggestion
```

I don't see the point of this tbh.

---
authors:
- jlpayne72: null
  name: Julian Payne
- davidrad: null
  name: David Radley  
date: "2026-04-13T08:00:00Z"
excerpt: Flink Community update for April 2026
title: Flink Community update for April 2026
aliases:
- /news/2026/04/13/community-update.html
---

This is the monthly Flink Community update for April 2026. We scour the latest updates from the Flink community so you don't have to! 

<!--more-->

We'll dive into a few hot topics and FLIPs (Flink Improvement Proposals) before providing the usual almanac-style listing of technical updates, user-facing updates, and governance and community.

<!-- TOC -->
  * [Hot Topics](#hot-topics)
  * [Developer/Technical Updates](#developertechnical-updates)
  * [User-Facing Updates](#user-facing-updates)
  * [Governance and Community](#governance-and-community)
<!-- TOC -->

## Hot Topics

### Apache Flink 2.3 Release Draws Nearer

Four months after the [release of Flink 2.2](https://flink.apache.org/2025/12/04/apache-flink-2.2.0-advancing-real-time-data--ai-and-empowering-stream-processing-for-the-ai-era/) (December 4th 2025), we are hotly anticipating the release of Flink 2.3. 

> The code freeze for Apache Flink 2.3 is planned for April 14th and at time of writing we anticipate the version to be launched toward the end of April 2026. 

Reviewer 3: it's now 1st may. "April 14th" should be past tense, and "end of April" has been and gone with no release.

Reviewer: good point!

### Checkpointing & State

Unaligned checkpoint recovery gained a new configuration option ([FLINK-38541](https://issues.apache.org/jira/browse/FLINK-38541)), giving operators finer control over recovery behaviour under failure conditions. Session-mode HA recovery has been enhanced with [FLINK-38975](https://issues.apache.org/jira/browse/FLINK-38975), introducing new Application stores for running and terminated applications.

### Metrics & Observability

gRPC batch export support was added to the metrics pipeline ([FLINK-39126](https://issues.apache.org/jira/browse/FLINK-39126)), providing a more robust export using gRPC for jobs with large numbers of operators or tasks.

### Python & Multi-language Support

Python users gained support for multiple compression formats in `python.files` ([FLINK-39260](https://issues.apache.org/jira/browse/FLINK-39260)), expanding deployment flexibility for Python UDF bundles.

> ## User-Facing Updates

Reviewer 3: this sectino heading confuses me. above, we talk about SQL changes in Materialized Tables, which is surely "User-facing", no? 
I think it's a spurious distinction tbh. Givne that the only entries under it are new releases, why not just call it "new releases" and ditch the "Developer/Technical Updates" heading?

Reviewer: @rmoff I think what we are doing each month is fitting the titles around the content that we have, rather than keeping a consistent format. I am ok with that, as we then do not have empty of scant sections. I do not have strong views on this. Maybe after more blog updates we will settle on meaningful sections, lets see how the content goes.

> ### Apache Flink CDC 3.6.0 Release

Reviewer 3: I don't understand why we're duplicating this https://github.com/apache/flink-web/pull/839/changes#diff-319d828b30adaaecfbe24565a9ab09a1dd74b76052fdc1cee1ed8218d61cd63fR39

---
authors:
- jlpayne72: null
  name: Julian Payne
date: "2026-04-13T08:00:00Z"
excerpt: Flink Community update for April 2026
title: Flink Community update for April 2026
aliases:
- /news/2026/04/13/community-update.html
---

This is the monthly Flink Community update for April 2026. We scour the latest updates from the Flink community so you don't have to! 
We'll dive into a few hot topics and FLIPs (Flink Improvement Proposals) before providing the usual almanac-style listing of technical updates, user-facing updates, and governance and community.
<!--more-->

> <!-- TOC -->

Reviewer 2: out of curiosity: what is the intention to have these comments here?

    * [Flink CDC 3.6.0 Release](#flink-cdc-360-release)
    * [Flink Documentation Re-structure](#flink-documentation-re-structure)
    * [Flink Native S3 FileSystem Connector](#flink-native-s3-filesystem-connector)
  * [Developer/Technical Updates](#developertechnical-updates)
    * [Table API & SQL](#table-api--sql)
    * [Runtime & Execution](#runtime--execution)
    * [Checkpointing & State](#checkpointing--state)
    * [Metrics & Observability](#metrics--observability)
    * [Python & Multi-language Support](#python--multi-language-support)
  * [User-Facing Updates](#user-facing-updates)
    * [Apache Flink CDC 3.6.0 Release](#apache-flink-cdc-360-release-1)
    * [Apache Flink Agents 0.2.1 Release](#apache-flink-agents-021-release)
    * [Documentation](#documentation)
  * [Governance and Community](#governance-and-community)
    * [FLIP Activity](#flip-activity)
      * [Accepted - being implemented](#accepted---being-implemented)
      * [Being discussed](#being-discussed)
    * [Apache Flink Dev List activity](#apache-flink-dev-list-activity)
  * [For more information](#for-more-information)

> <!-- TOC -->

Reviewer 2: same about comment

Previous Blog: [Flink community update March 2026](https://flink.apache.org/2026/03/01/flink-community-update-for-march-2026/)

## Hot Topics

### Apache Flink 2.3 Release Draws Nearer

> Four months after the release of Flink 2.2 (December 4th 2025), we are hotly anticipating the release of Flink 2.3. 

Reviewer 3: ```suggestion
Four months after the [release of Flink 2.2](https://flink.apache.org/2025/12/04/apache-flink-2.2.0-advancing-real-time-data--ai-and-empowering-stream-processing-for-the-ai-era/) (December 4th 2025), we are hotly anticipating the release of Flink 2.3. 
```

The code freeze for Apache Flink 2.3 was invoked on April 14th and at time of writing we anticipate the version to be launched toward the end of April 2026. 

> The release will be subject to its own launch notices and blog, which we will not repeat here. Needless to say we're excited as the release scope includes important updates to Materialized Tables for Flink SQL, OTel gRPC exporter and watermark alignment for backlogged jobs in Runtime, and in Connectors, an update to the S3 Filesink connector. 

Reviewer 3: Maybe this is just me, but everyone is so _excited_ about _everything_.
Just like `leverage`, this word has become meaningless/devalued. _Personally_ (and it is very much IMHO) I would go for something more like:

```suggestion
Some of the main features in this release will include important updates to Materialized Tables for Flink SQL, OTel gRPC exporter and watermark alignment for backlogged jobs in Runtime, and in Connectors, an update to the S3 Filesink connector. 
Stay tuned to this blog for the release announcement with full details.
```

also moved the note about separate blog post to the end of the para

> The release will be subject to its own launch notices and blog, which we will not repeat here. Needless to say we're excited as the release scope includes important updates to Materialized Tables for Flink SQL, OTel gRPC exporter and watermark alignment for backlogged jobs in Runtime, and in Connectors, an update to the S3 Filesink connector. 

Reviewer: I am not sure what OTel gRPC exporter is - a link to the detials would help and an explanation here as to why this is valuable and what it enables you to do

> The release will be subject to its own launch notices and blog, which we will not repeat here. Needless to say we're excited as the release scope includes important updates to Materialized Tables for Flink SQL, OTel gRPC exporter and watermark alignment for backlogged jobs in Runtime, and in Connectors, an update to the S3 Filesink connector. 

Reviewer: detailing the current issues and or the value of watermark alignment for backlogged jobs would be a good way to get people interested and know either it is relevant to them.

> We are also proposing a re-structure to the Flink documentation to make it easier to navigate for new and existing users. 

Reviewer 3: is this coupled to 2.3? If not, cut this line. If so, include that note in the related para later on.

> You can see the full release scope here: [https://cwiki.apache.org/confluence/display/FLINK/2.3+Release](https://cwiki.apache.org/confluence/display/FLINK/2.3+Release)

Reviewer 3: ```suggestion
You can see the full release scope [here](https://cwiki.apache.org/confluence/display/FLINK/2.3+Release).
```

hyperlinks, not URLs :)

### Flink CDC 3.6.0 Release

Flink CDC is used to capture and stream real-time changes from databases (inserts, updates, and deletes) as they happen. 

> If you missed it, you can read the [launch announcement](https://flink.apache.org/2026/03/30/apache-flink-cdc-3.6.0-release-announcement/). 

Reviewer 3: ```suggestion
```

> Version 3.6.0 extends Flink version support to 1.20.x and 2.2.x, upgrades JDK version to 11, introduces new Oracle Source and Apache Hudi Sink Pipeline connectors and adds Lenient mode schema evolution support for Fluss Pipeline connector. 

Reviewer 3: ```suggestion
[Version 3.6.0](https://flink.apache.org/2026/03/30/apache-flink-cdc-3.6.0-release-announcement/) extends Flink version support to 1.20.x and 2.2.x, upgrades JDK version to 11, introduces new Oracle Source and Apache Hudi Sink Pipeline connectors and adds Lenient mode schema evolution support for Fluss Pipeline connector. 
```

> Version 3.6.0 extends Flink version support to 1.20.x and 2.2.x, upgrades JDK version to 11, introduces new Oracle Source and Apache Hudi Sink Pipeline connectors and adds Lenient mode schema evolution support for Fluss Pipeline connector. 

Reviewer: it would be great highlight the business value of each of these things, with hyper links for each of them.

> It also introduces PostgreSQL Schema Evolution support, enhances Schema Evolution capabilities for better multi-table synchronization scenarios and table name mapping flexibility, and strengthens the Transform framework with VARIANT type and JSON parsing support.

Reviewer: enhances ->  enhancing

### Flink Documentation Re-structure

We want to re-structure Flink documentation to aid discoverability and reduce duplication (see [FLIP-561](https://cwiki.apache.org/confluence/display/FLINK/FLIP-561%3A+Restructure+Flink+documentation)). 
Reflecting the broadening use of SQL we plan to create a dedicated Flink SQL section separate from Table API. 
We will move shared concepts (Relational Streaming concepts like Dynamic Tables, Time Attributes) to the top-level Concepts section and also move shared architecture documentation (Source/Sink APIs) to the Connectors section. 

> Additionally, Python documentation will now be integrated within the Table API and DataStream API sections where applicable, with the rest moved to the Python API docs (PyDocs).

Reviewer: maybe describe what` the rest` means

### Flink Native S3 FileSystem Connector

Amazon S3 is a popular sink and source destination for Apache Flink users. 
Currently, Apache Flink provides two primary mechanisms for interacting with S3 (`flink-s3-fs-hadoop` and `flink-s3-fs-presto`), both of which are adapters wrapping external projects. 
To improve performance and maintainability we propose the creation of `flink-s3-fs-native` ([FLIP-555](https://cwiki.apache.org/confluence/display/FLINK/FLIP-555%3A+Flink+Native+S3+FileSystem)), a clean-slate implementation built directly on the AWS SDK for Java v2. 
We will remove the Hadoop dependency, making the connector a self-contained module depending only on the modular AWS SDK v2. 
The connector will support both state access (FileSystem) and transactional sinks (RecoverableWriter), simplifying the user experience. Additionally, the connector leverages the non-blocking I/O capabilities of Netty and the AWS Common Runtime (CRT) to maximise throughput.

## Developer/Technical Updates

### Table API & SQL

> Continued work on Materialized Tables including `START_MODE` clause parser support ([[FLINK-39304]](https://issues.apache.org/jira/browse/FLINK-39304), [[#27843]](https://github.com/apache/flink/pull/27843)), interval reuse for `FRESHNESS` and `START MODE` ([[#27845]](https://github.com/apache/flink/pull/27845)), and `CREATE OR ALTER` now correctly respects existing table schemas ([[FLINK-39284]](https://issues.apache.org/jira/browse/FLINK-39284), [[#27854]](https://github.com/apache/flink/pull/27854)). `TO_CHANGELOG` retract/upsert stream conversion was also added ([[#27847]](https://github.com/apache/flink/pull/27847)).

Reviewer 3: Up to you, but I don't find this particularly readable with the interleaving of JIRAs  and GH issue codes. 
I would like to one, not both (probably just the JIRA, unless there is only GH). 
Also get rid of the double parentheses `([])` which seems redundant.

> BITMAP Type support was introduced end-to-end: the core data type ([[FLINK-38852]](https://issues.apache.org/jira/browse/FLINK-38852), [[FLINK-39185]](https://issues.apache.org/jira/browse/FLINK-39185)), scalar functions ([[FLINK-39186]](https://issues.apache.org/jira/browse/FLINK-39186)), aggregate functions ([[FLINK-39187]](https://issues.apache.org/jira/browse/FLINK-39187)), and documentation ([[FLINK-39188]](https://issues.apache.org/jira/browse/FLINK-39188), [[#27835]](https://github.com/apache/flink/pull/27835)).

Reviewer 2: ```suggestion
`BITMAP` Type support was introduced end-to-end: the core data type ([[FLINK-38852]](https://issues.apache.org/jira/browse/FLINK-38852), [[FLINK-39185]](https://issues.apache.org/jira/browse/FLINK-39185)), scalar functions ([[FLINK-39186]](https://issues.apache.org/jira/browse/FLINK-39186)), aggregate functions ([[FLINK-39187]](https://issues.apache.org/jira/browse/FLINK-39187)), and documentation ([[FLINK-39188]](https://issues.apache.org/jira/browse/FLINK-39188), [[#27835]](https://github.com/apache/flink/pull/27835)).
```

> BITMAP Type support was introduced end-to-end: the core data type ([[FLINK-38852]](https://issues.apache.org/jira/browse/FLINK-38852), [[FLINK-39185]](https://issues.apache.org/jira/browse/FLINK-39185)), scalar functions ([[FLINK-39186]](https://issues.apache.org/jira/browse/FLINK-39186)), aggregate functions ([[FLINK-39187]](https://issues.apache.org/jira/browse/FLINK-39187)), and documentation ([[FLINK-39188]](https://issues.apache.org/jira/browse/FLINK-39188), [[#27835]](https://github.com/apache/flink/pull/27835)).

Reviewer: you could add - the value of BITMAP is ....

> Join Operations saw cascaded delta join support merged ([[FLINK-39233]](https://issues.apache.org/jira/browse/FLINK-39233), [[#27828]](https://github.com/apache/flink/pull/27828)), lookup join after delta join ([[FLINK-39174]](https://issues.apache.org/jira/browse/FLINK-39174)), and immutable column sink mode traits ([[FLINK-39287]](https://issues.apache.org/jira/browse/FLINK-39287), [[#27838]](https://github.com/apache/flink/pull/27838)).

Reviewer: maybe say - allowing the delta join to be used in more scenarios.

> Code generation and correctness work included: NDU analyzer improvements for non-deterministic function detection ([[FLINK-39313]](https://issues.apache.org/jira/browse/FLINK-39313), [[#27819]](https://github.com/apache/flink/pull/27819)), duplicate function instance elimination ([[FLINK-39094]](https://issues.apache.org/jira/browse/FLINK-39094)), filter optimisation for upsert key groups ([[FLINK-39314]](https://issues.apache.org/jira/browse/FLINK-39314)), a UDF lambda expression argument fix ([[#27787]](https://github.com/apache/flink/pull/27787)), and JSON field name code generation fixes backported across multiple releases ([[#27850]](https://github.com/apache/flink/pull/27850), [[#27851]](https://github.com/apache/flink/pull/27851), [[#27856]](https://github.com/apache/flink/pull/27856)).

Reviewer 2: looks like here and in other places we have a mix of jira issue links and GH PRs...
jira should have more end user friendly explanation while PRs should have more deep dive implementation related things... 
I wonder if here we should use only jira links

### Runtime & Execution

> Adaptive scheduler and `ExecutionGraph` improvements will be part of Flink 2.3. Rescale history now records statistics ([[#27540]](https://github.com/apache/flink/pull/27540)) and rescale information ([[#27539]](https://github.com/apache/flink/pull/27539)). A configurable `ExecutionGraph` cache TTL was introduced ([[#27509]](https://github.com/apache/flink/pull/27509)), and the Flink Web UI was updated to surface rescale and configuration data for adaptive scheduler jobs ([[#27826]](https://github.com/apache/flink/pull/27826)).

Reviewer: a notice the tenses change. There will be improvements in 2.3. Rescale history now records, and the cache TTL was introduced and UI was updated.

### Checkpointing & State

> Unaligned checkpoint recovery gained a new configuration option ([[FLINK-38541]](https://issues.apache.org/jira/browse/FLINK-38541), [[#27782]](https://github.com/apache/flink/pull/27782)), giving operators finer control over recovery behaviour under failure conditions. Session-mode HA recovery re-execution was also resolved ([[FLINK-38975]](https://issues.apache.org/jira/browse/FLINK-38975)).

Reviewer: what does was resolved means in the sentence " Session-mode HA recovery re-execution was also resolved". I am not sure how to read this , is it fixing something or introducing something

### Metrics & Observability

gRPC batch export support was added to the metrics pipeline ([[#27692]](https://github.com/apache/flink/pull/27692)), providing a more robust export using gRPC for jobs with large numbers of operators or tasks.

### Python & Multi-language Support

Python users gained support for multiple compression formats in `python.files` ([[#27780]](https://github.com/apache/flink/pull/27780)), expanding deployment flexibility for Python UDF bundles.

> ## User-Facing Updates

Reviewer 2: I didn't get the difference between 
>## Developer/Technical Updates

and 
>## User-Facing Updates

especially given the fact that 
>### Table API & SQL

in inside the first one

can you elaborate then where the boundary is then?

Reviewer 2: imho: `## User-Facing Updates` probably should be renamed to something like 
`## Release Updates`

### Apache Flink CDC 3.6.0 Release

- **Announcement:** [Apache Flink CDC 3.6.0 Release Announcement](https://flink.apache.org/2026/03/30/apache-flink-cdc-3.6.0-release-announcement/) — *Yanquan Lv, March 30*
- **Release Manager:** Yanquan Lv
- **Key highlights:**
  - Extended Flink version support to 1.20.x and 2.2.x
  - Upgraded minimum JDK to version 11
  - New connectors added
  - Expanded schema evolution capabilities
  - PostgreSQL pipeline connector schema change support [[FLINK-38959]](https://issues.apache.org/jira/browse/FLINK-38959)

>   - Oracle pipeline support for column nullable changes [[FLINK-39196]](https://issues.apache.org/jira/browse/FLINK-39196)

Reviewer 3: doesn't this just duplicate what you have above under "hot topics" ?
why does CDC get "hot topic" treatment but agents not?

I'd pick one section and stick with it for both of these. Probably, "New Releases" ("User-Facing Updates" is kinda wordy).

### Apache Flink Agents 0.2.1 Release

- **Announcement:** [Apache Flink Agents 0.2.1 Release Announcement](https://flink.apache.org/2026/03/26/apache-flink-agents-0-2-1-release-announcement/) — *Wenjin Xie, March 26*
- **Release Manager:** Wenjin Xie
- **Key highlights:**
  - 3 targeted bug fixes
  - Vulnerability fixes
  - Minor improvements

### Documentation

March saw a documentation push across several areas:
* Materialized tables `CREATE`/`ALTER` operations [[#27846]](https://github.com/apache/flink/pull/27846)
* SQL function consistency improvements [[#27842]](https://github.com/apache/flink/pull/27842)
* BITMAP type coverage [[#27835]](https://github.com/apache/flink/pull/27835)
* PostgreSQL CDC `schema.change.enabled` parameter [[FLINK-39294]](https://issues.apache.org/jira/browse/FLINK-39294)
* MySQL 8.4+ support [[FLINK-39295]](https://issues.apache.org/jira/browse/FLINK-39295)
* MySQL connector consistency fixes [[FLINK-39344]](https://issues.apache.org/jira/browse/FLINK-39344)

> 

Reviewer 3: is there anything of _note_ in these?

## Governance and Community

> ### FLIP Activity

Reviewer 3: ```suggestion
### FLIP (accepted)
```

Reviewer 3: consider adding one sentence, perhaps above this heading, explaining what a FLIP is.

> #### Accepted - being implemented

Reviewer 3: ```suggestion
```

**[FLIP-561](https://cwiki.apache.org/confluence/display/FLINK/FLIP-561%3A+Restructure+Flink+documentation): Restructure Flink Documentation.** (Piotr Nowojski)
- A major reorganisation to improve discoverability and reduce duplication across the Flink docs. Updated March 24, 2026.

**[FLIP-558](https://cwiki.apache.org/confluence/display/FLINK/FLIP-558%3A+Improvements+to+SinkUpsertMaterializer+and+changelog+disorder): Improvements to SinkUpsertMaterializer and Changelog Disorder.**
- Addresses correctness issues in upsert materialisation under out-of-order changelog events.

**[FLIP-557](https://cwiki.apache.org/confluence/display/FLINK/FLIP-557%3A+Granular+Control+over+Data+Reprocessing+in+Materialized+Table+Evolution): Granular Control over Data Reprocessing in Materialized Table Evolution.**
- Gives users finer control over which partitions are reprocessed when a materialized table definition changes.

**[FLIP-551](https://cwiki.apache.org/confluence/display/FLINK/FLIP-551%3A+Make+FRESHNESS+Optional+for+Materialized+Tables): Make FRESHNESS Optional for Materialized Tables.** (targeting Flink 2.2)

> - Reduces the boilerplate required when defining materialized tables where freshness guarantees are not needed.

Reviewer 3: ```suggestion
* **[FLIP-561](https://cwiki.apache.org/confluence/display/FLINK/FLIP-561%3A+Restructure+Flink+documentation): Restructure Flink Documentation.** (Piotr Nowojski) - A major reorganisation to improve discoverability and reduce duplication across the Flink docs. Updated March 24, 2026.
* **[FLIP-558](https://cwiki.apache.org/confluence/display/FLINK/FLIP-558%3A+Improvements+to+SinkUpsertMaterializer+and+changelog+disorder): Improvements to SinkUpsertMaterializer and Changelog Disorder.** - Addresses correctness issues in upsert materialisation under out-of-order changelog events.
* **[FLIP-557](https://cwiki.apache.org/confluence/display/FLINK/FLIP-557%3A+Granular+Control+over+Data+Reprocessing+in+Materialized+Table+Evolution): Granular Control over Data Reprocessing in Materialized Table Evolution.** - Gives users finer control over which partitions are reprocessed when a materialized table definition changes.
* **[FLIP-551](https://cwiki.apache.org/confluence/display/FLINK/FLIP-551%3A+Make+FRESHNESS+Optional+for+Materialized+Tables): Make FRESHNESS Optional for Materialized Tables.** (targeting Flink 2.2) - Reduces the boilerplate required when defining materialized tables where freshness guarantees are not needed.
```

Reviewer 3: "targeting Flink 2.2" seems a bit out of date?

Reviewer 2: I wonder why FLIP-551 appeared here at all...
it was delivered with 2.2.0

> #### Being discussed

Reviewer 3: ```suggestion
### FLIPs (under discussion)
```

**[FLIP-202](https://cwiki.apache.org/confluence/display/FLINK/FLIP-202): [DRAFT] Introduce ClickHouse Connector.**
- Community discussion ongoing for an officially supported ClickHouse sink connector.

**[FLIP-332](https://cwiki.apache.org/confluence/display/FLINK/%5BWIP%5DFLIP-332%3A+Introduce+the+concept+of+state+self-sustained): [WIP] Introduce State Self-Sustained Concept.**
- Explores state backends capable of managing their own lifecycle independently of checkpoints.

**[FLIP-267](https://cwiki.apache.org/confluence/display/FLINK/FLIP+267%3A+Iceberg+Connector): Iceberg Connector.**

> - Continuing effort to formalise the Iceberg integration as a first-class Flink connector.

Reviewer 3: ```suggestion
* **[FLIP-202](https://cwiki.apache.org/confluence/display/FLINK/FLIP-202): [DRAFT] Introduce ClickHouse Connector.** - Community discussion ongoing for an officially supported ClickHouse sink connector.
* **[FLIP-332](https://cwiki.apache.org/confluence/display/FLINK/%5BWIP%5DFLIP-332%3A+Introduce+the+concept+of+state+self-sustained): [WIP] Introduce State Self-Sustained Concept.** - Explores state backends capable of managing their own lifecycle independently of checkpoints.
* **[FLIP-267](https://cwiki.apache.org/confluence/display/FLINK/FLIP+267%3A+Iceberg+Connector): Iceberg Connector.** - Continuing effort to formalise the Iceberg integration as a first-class Flink connector.
```

### Apache Flink Dev List activity

> You can view the dev list archives [online](https://lists.apache.org/list.html?dev@flink.apache.org) or [subscribe](https://flink.apache.org/community.html#mailing-lists) to receive emails.

Reviewer 3: dup of lines 174-176

```suggestion
```

> ## For more information

Reviewer 3: ```suggestion
## Staying up to date
```

Follow this blog to keep up to date with what is happening in the Flink community.

If you have ideas for what you would like to see in these blogs or there is anything you think has been misrepresented, is wrong or missing, please let us know via the dev list (detail below) or [Slack](https://flink.apache.org/how-to-contribute/getting-help/#slack).

If you would like to keep a closer eye on what's happening in the community, subscribe to one of the [Flink community mailing lists](https://flink.apache.org/community.html#mailing-lists) to get fine-grained weekly updates, upcoming event announcements and more.
Two popular mailing lists are:
* the [dev list](https://lists.apache.org/list.html?dev@flink.apache.org) for development related discussions

> * the [user list](https://lists.apache.org/list.html?user@flink.apache.org) for user support and questions

Reviewer 3: ```suggestion
There are several ways that you can keep up to date with what is happening in the Flink community.
For a full list see [here](https://flink.apache.org/what-is-flink/community/).

* Follow this blog.
An [RSS feed](https://flink.apache.org/posts/index.xml) is available.
* Subscribe to one of the [Flink community mailing lists](https://flink.apache.org/community.html#mailing-lists).
Two popular mailing lists are:
    * the [dev list](https://lists.apache.org/list.html?dev@flink.apache.org) for development related discussions
    * the [user list](https://lists.apache.org/list.html?user@flink.apache.org) for user support and questions
* Join the [Apache Flink Slack group](https://flink.apache.org/what-is-flink/community/#slack).
```
